.github/workflows/pipeline.yaml

```diff
@@ -184,13 +209,13 @@ jobs:
             --template packaged-testing.yaml \
             --capabilities CAPABILITY_IAM \
             --region ${TESTING_REGION} \
-            --s3-bucket ${TESTING_ARTIFACTS_BUCKET} \
-            --image-repository ${TESTING_IMAGE_REPOSITORY} \
+            --s3-bucket ${{ secrets.TESTING_ARTIFACTS_BUCKET }} \
+            --image-repository ${{ secrets.TESTING_IMAGE_REPOSITORY }} \
             --no-fail-on-empty-changeset \
-            --role-arn ${TESTING_CLOUDFORMATION_EXECUTION_ROLE}
+            --role-arn ${{ secrets.TESTING_CLOUDFORMATION_EXECUTION_ROLE }}
 
   integration-test:
```

Comment:
Tests should be separate from deploy. 

So you should have 

Test Pipeline
Deploy-dev pipeline
Deploy-prod pipeline 

You can then disallow deploy to run if the dev fails - sounds a bit more easy to understand for anyone looking at your repo for the first time?